autoload/asyncomplete.vim

```diff
@@ -278,9 +278,8 @@ function! s:remote_refresh(ctx, force) abort
         let l:typed_len = l:endpos - l:startpos
         if l:typed_len == 1
             call add(l:sources_to_notify, l:name)
-        elseif has_key(s:matches, l:name) && s:matches[l:name]['refresh']
-            call add(l:sources_to_notify, l:name)
-        elseif s:supports_smart_completion()
+        endif
+        if has_key(s:matches, l:name) && s:matches[l:name]['refresh']
```

Comment:
This one refresh all the sources. We should only refresh the source that has `refresh=1`. can you try with multiple sources and check the logs.

Reply:
I think this is because of `if l:typed_len == 1`.  What is the purpose of this `if` block?